src/topp/MzTabExporter.cpp

```diff
@@ -437,38 +514,87 @@ namespace OpenMS
           const std::vector<ProteinIdentification::ProteinGroup> indist_groups = it->getIndistinguishableProteins();
           const std::vector<ProteinHit> protein_hits = it->getHits();
 
+          // pre-analyze data for occuring meta values at protein hit level
+          // these are used to build optional columns containing the meta values in internal data structures
+          set<String> protein_hit_user_value_keys;
+          for (Size i = 0; i < protein_hits.size(); ++i)
```

Comment:
looks a bit like the helper code chris wrote. would it be possible to reuse it or move your code there?